# Do not use deprecated jQuery.sap API (no-jquery-sap)

This rule shall ensure that no deprecated jQuery.sap API is used


## Rule Details

The UI5 developers work hard to remove jQuery from the library. Since version  1.58

Examples of **incorrect** code for this rule:

```js

// fill me in

```

Examples of **correct** code for this rule:

```js

// fill me in

```

### Options

If there are any options, describe them here. Otherwise, delete this section.

## When Not To Use It

Give a short description of when it would be appropriate to turn off this rule.

## Further Reading

If there are other links that describe the issue this rule addresses, please include them here in a bulleted list.
